frame/elections-phragmen/src/lib.rs

```diff
@@ -258,46 +297,67 @@ decl_module! {
 		const TermDuration: T::BlockNumber = T::TermDuration::get();
 		const ModuleId: LockIdentifier  = T::ModuleId::get();
 
-		/// Vote for a set of candidates for the upcoming round of election.
+		/// Vote for a set of candidates for the upcoming round of election. This can be called to
+		/// set the initial votes, with `would_create = true`, or update already existing votes
+		/// with `would_create = false`.
+		///
+		/// Upon initial voting, `value` units of `who`'s balance is locked and a bond amount is
+		/// reserved.
 		///
 		/// The `votes` should:
 		///   - not be empty.
-		///   - be less than the number of candidates.
+		///   - be less than the number of possible candidates. Note that all current members and
+		///     runners-up are also automatically candidates for the next round.
 		///
-		/// Upon voting, `value` units of `who`'s balance is locked and a bond amount is reserved.
 		/// It is the responsibility of the caller to not place all of their balance into the lock
 		/// and keep some for further transactions.
 		///
 		/// # <weight>
-		/// #### State
-		/// Reads: O(1)
-		/// Writes: O(V) given `V` votes. V is bounded by 16.
+		/// Base weight: if would_create { 47.93 µs } else { 38.46 µs }
+		/// State reads:
+		/// 	- Candidates.len() + Members.len() + RunnersUp.len()
+		/// 	- Voting (is_voter)
+		/// 	- [AccountBalance(who) (unreserve + total_balance)]
+		/// State writes:
+		/// 	- Voting
+		/// 	- Lock
+		/// 	- [AccountBalance(who) (unreserve -- only when would_create is true)]
 		/// # </weight>
-		#[weight = 100_000_000]
-		fn vote(origin, votes: Vec<T::AccountId>, #[compact] value: BalanceOf<T>) {
+		#[weight = if *would_create {
+			50 * WEIGHT_PER_MICROS + T::DbWeight::get().reads_writes(4, 2)
+		} else {
+			40 * WEIGHT_PER_MICROS + T::DbWeight::get().reads_writes(4, 2)
+		}]
+		fn vote(
+			origin,
+			votes: Vec<T::AccountId>,
+			#[compact] value: BalanceOf<T>,
+			would_create: bool,
```

Comment:
big no here imo!

You have introduced some parameter to a function to save on 10 microseconds?

This is def not right

Reply:
yeah I was perhaps trying to be too pedantic here. You're right this is not significant and will revert it.